[TSK-885] [TSK-886] ui-kit: UserGrid

ObitoDevv opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `User Grid`
- Figma link or screenshot: [Figma Design](https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=7252-332368&t=Z9yOThOzKCxx7IT2-4)

---


## React Best Practices Checklist

- [x] I have avoided **unnecessary re-renders** by using composition (`children` as props), whenever possible. This applies, in particular, to composite components (like lists of components, and the like).
- [x] **I have designed my components as server-side whenever possible**—I have not used `useState`, `useEffect`, `useContext`, or any other React hook that works with "use client" without a good reason.
- [x] **I have not put props in state**—in general, props should remain as props unless the component needs an `initialValue`, and judiciously manage its own independent state detached from the parent component.
    - If I have stored props in state, I have explained where exactly and justified it with a solid reason here below:
    
- [x] **I have avoided using Context** in any critical component in terms of performance
    - If I have used Context, I have explained where exactly and justified it with a solid reason here below:


## General Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.




---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #131  (issue)

alebg commented:
Hi @ObitoDevv , as discussed in the call, please integrate the icons I send to @anilrise11 in the UserGrid, as per this screenshot (made by @carlapbg ):

![immagine](https://github.com/user-attachments/assets/5cf5d21e-9534-4e97-b502-9d8593bdf5c6)

Please note which icon corresponds to which role. Also, you can modify slighltly the user props such that you always have objects `roles: {platformName: string; role: string}[]` which you can use to render the platform name + icon

carlapbg reviewed (changes requested):
When selected, the icon should have the same color as the text. Both should change the same

<img width="252" alt="Screenshot 2025-05-15 at 22 42 26" src="https://github.com/user-attachments/assets/9df51406-f4bc-4f11-a5fe-0dcc9ea9aed0" />


Also, is it possible to add a "Clear all filters" button? Something like this:
![Screenshot 2025-05-15 at 22 55 45](https://github.com/user-attachments/assets/eaf8a84b-2025-41ee-8e6e-669636c28853)

ObitoDevv commented:
@carlapbg Hi mam, I have fixed the tab color issue and added the `clear filter` button to clear all filters at once. 

Can you please verify the changes ?

MytsV commented:
I fixed all of the small issues I found. The changes I requested personally seem to be addressed in a different PR.

A few notes:
- there might be a need to indicate the number of filters applied like in Figma
- there's some flicker of the button's color that I see when reloading the page. can't pinpoint the reason for it yet, will have to see in the integration

alebg reviewed (dismissed):
Thanks a lot @ObitoDevv @MytsV !

@ObitoDevv @anilrise11 See my last commit: note that I refactored a couple of components that were missing localization. Please also take note of the changes made by Viktoriia and me on the UserGrid

alebg commented on packages/ui-kit/lib/components/grids/user-grid.tsx:
```diff
@@ -70,69 +186,487 @@ export const UserGrid = (props: UserGridProps) => {
-    </div>;
+
+        if (props.onSendNotifications) {
+            alert(dictionaryFormat(dictionary.sendingNotifications, { count: selectedUserIds.length }));
```
@maany I added a small util in the translations package to allow me to mimick Python's format strings and do this. This entry in the dictionary is `      sendingNotifications: 'Sending notifications to %{count} selected users',`  and  this util function replaces `%{count}` with the dynamic value.

Downside of the util is that it's not typed or anything, so nothing fails if you don't pass the format keys needed, or if you pass a key that is not present. TBD if this is a good util to have, and if so, how to improve it.

maany replied:
You can do this in js with 
```js
export default const Component = (props: {count: number}) => {
  return (
    <div>
        `Sending notifications to ${props.count}`
    </div> 
}
```

maany replied:
I see, we would need something like liquid js in the translations package in that case. https://liquidjs.com/
